src/stream/src/executor/watermark_filter.rs

```diff
@@ -253,7 +254,21 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
                         .await?;
                     }
 
-                    if is_checkpoint {
+                    if let Some(mutation) = barrier.mutation.as_deref() {
+                        match mutation {
+                            Mutation::Pause => {
+                                is_paused = true;
+                            }
+                            Mutation::Resume => {
+                                is_paused = false;
+                            }
+                            _ => (),
+                        }
+                    }
+
+                    yield Message::Barrier(barrier);
```

Comment:
I guess this should probably continue to be done before `Self::get_global_max_watermark`, otherwise waiting on `Committed(prev_epoch)` may lead to deadlock.